### DISCUSSION TITLE
Problem with $user->syncGroups

### DISCUSSION THREAD

**Discussion by ondequadre** [codeigniter4/shield#277] [Q&A] [answered]
I don't know exactly if I am doing something wrong or there is a problem with syncGroups()

If I do this:

```php
$groups = "'" . implode("', '", $this->request->getPost('groups')) . "'";
$user->syncGroups($groups);
```

Returns this error:

```
CodeIgniter\Shield\Authorization\AuthorizationException #401
'admin', 'user' is not a valid group. 
```

if I manually insert the groups:

```php
$user->syncGroups('admin','user');
```

then it works...

**Comment by kenjis** [comment]
It is true, `'admin', 'user'` is not a valid group.

See https://github.com/codeigniter4/shield/blob/develop/docs/quickstart.md#adding-a-group-to-a-user

**Comment by ondequadre** [comment]
Thanks kenjis, but what I'm trying to do is not add a group to the user while creating. But I'm trying to change groups for a user while editing. In fact if I write the groups dynamically it gives me the error, but if I write the command manually in the code it works.

This does not work:

```php
            $data = [
                'firstname' => $this->request->getPost('firstname'),
                'lastname' => $this->request->getPost('lastname'),
                'email' => $this->request->getPost('email'),
                'company' => $this->request->getPost('company')
            ];

            $user  = $this->userModel->findById($id);

            $user->fill($data);

            if ($this->userModel->save($user))
            {
                $groups = "'" . implode("', '", $this->request->getPost('groups')) . "'";

                $user->syncGroups($groups);
            }
```
```
CodeIgniter\Shield\Authorization\AuthorizationException #401
'admin', 'user' is not a valid group. 
```

This works:

```php
            $data = [
                'firstname' => $this->request->getPost('firstname'),
                'lastname' => $this->request->getPost('lastname'),
                'email' => $this->request->getPost('email'),
                'company' => $this->request->getPost('company')
            ];

            $user  = $this->userModel->findById($id);

            $user->fill($data);

            if ($this->userModel->save($user))
            {
                $user->syncGroups('admin', 'user');
            }
```

**Reply by kenjis** [reply]
See the docs.
You need to pass string params to `syncGroups()`:
```php
$user->syncGroups('admin', 'beta');
```
But you pass string `"'admin', 'user'"`. So it does not work.

**Comment by ondequadre** [comment]
Do you have any suggestions? Because I have tried several solutions I still have not been able to solve.

**Reply by kenjis** [reply]
See https://www.php.net/manual/en/migration56.new-features.php#migration56.new-features.splat

**Comment by ondequadre** [comment, marked as answer]
Thank you kenjis I solved it, I admit it was something completely new to me, I didn't know this possibility existed.

`$user->syncGroups(...$this->request->getPost('groups'));`
